tools/deploy-qemu.py

```diff
@@ -0,0 +1,71 @@
+import os
+import shutil
+import subprocess
+import sys
+import tempfile
+
+
+HELP_TEXT="""deploy-qemu IMAGE USERDATA
+
+Starts an ephemeral virtual machine in qemu, injecting configuration via
+cloud-init. Stopping this script stops the VM and discards all data.
+
+IMAGE    -- An os image that can be booted by qemu and has cloud-init
+            installed and enabled. No changes are made to this file.
+
+USERDATA -- A cloud-init user-data config file, or a directory of
+            configuration as accepted by the `gen-user-data` tool.
+
+In addition, if the QEMU_EXTRA_ARGS environment variable is defined, it adds
+its content as additional arguments to qemu."""
+
+
+if len(sys.argv) != 3:
+    print(HELP_TEXT)
+    exit(1)
+
+script_dir = os.path.dirname(os.path.realpath(__file__))
+image = sys.argv[1]
+userdata = sys.argv[2]
+
+qemu_extra_args = os.getenv("QEMU_EXTRA_ARGS")
+qemu_extra_args = qemu_extra_args.split(' ') if qemu_extra_args is not None else []
```

Comment:
I think you can drop this line and modify the previous one to:

```python
qemu_extra_args = os.getenv("QEMU_EXTRA_ARGS", [])
```

Reply:
It actually is more complicated. If you simply read the environment variable you could get sth like "-smp 2 -m 2048", but you need to split it to make it a list.

I've considered these two options, but rejected both:
 * This returns either a string or a list, so the processing will be difficult: `os.getenv("QEMU_EXTRA_ARGS", [])`
 * `os.getenv("QEMU_EXTRA_ARGS", "").split(' ')` returns `[""]` when QEMU_EXTRA_ARGS is empty

Eventually I decided to go with the solution in this patch.

Reply:
Oh dear, didn't read it properly. Thanks, makes sense!

Reply:
Technically, we'd probably have to use `shlex` here, but we didn't do this in the bash version either.

Reply:
Oh wow, I had no idea `shlex` exists. But I hope it is just fine like this.